### PR TITLE
Fix controller pointers with Solvers

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview
This change fixes issue with ability to select hand rays for solver tracking. There was the option to select a hand joint but the forward vector would point through the hand. Thus with a hand raised, it pointed up with gravity (relatively upwards to the sky) instead of straight out through the palm. 

This change modifies the `MotionController` enum type to be more extensible and accurate as the `ControllerRay` type. Instead of looking for some odd (easily breaking code) of a visual's transform on the motion controller, the code now looks for a particular line pointer available which will work for both motion controller and hand controllers.

Also, updated scene to use styling seen in other scenes as well as cleaner radial set prefab instead of just box buttons list.

image

SM_Fix

Also updates documentation and fixes some doc errors
image

## Changes
- Fixes: #6279

### CODE REVIEW

File: Documentation/README_Solver.md [hunk 1]
```diff
@@ -35,16 +35,16 @@ In order to use the Solver system, simply add one of the components listed above
 
 The *Tracked Target Type* property of the [`SolverHandler`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SolverHandler) component defines the point of reference all solvers will use to calculate their algorithms. For example, a value type of [`Head`](xref:Microsoft.MixedReality.Toolkit.Utilities.TrackedObjectType.Head) with a simple [`SurfaceMagnetism`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SurfaceMagnetism) component will result in a raycast from the head and in the direction of the user's gaze for solving what surface is hit. Potential values for the `TrackedTargetType` property are:
 
-* *Head* : Point of reference from the camera
-* *MotionController*: Point of reference from a motion controller
+* *Head* : Point of reference is the transform of the main camera
+* *ControllerRay*: Point of reference is the line pointer on a controller (i.e motion controller or hands)
```
Comment (Cameron-Micka) [inline Documentation/README_Solver.md]: Nit: I'm not quite sure if this is the beginning or end of a "line pointer" or what a "line pointer" is on a hand?

Comment (Troy-Ferrell) [reply Documentation/README_Solver.md]:
```suggestion
* *ControllerRay*: Point of reference is the [`LinePointer`](xref:Microsoft.MixedReality.Toolkit.Input.LinePointer) transform on a controller (i.e pointer origin on a motion controller or hand controller) pointing in the direction of the line ray
```

File: Documentation/README_Solver.md [hunk 2]
```diff
@@ -156,31 +156,36 @@ To force the associated GameObject to stay vertical in any mode other than *None
 
 ![SurfaceMagnetism Example](../Documentation/Images/Solver/SurfaceMagExample.png)
 
+#### Determining what surfaces can be hit
+
+When adding a [`SurfaceMagnetism`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SurfaceMagnetism) component to a GameObject, it is important to consider the layer of the GameObject and it's children, if any have colliders. The component works by performing various types of raycasts to determine what surface to "magnet" itself against. If the Solver GameObject has a collider on one of the layers listed in the `MagneticSurfaces` property of `SurfaceMagnetism`, then the raycast will likely hit itself resulting in the GameObject attaching to it's own collider point. This odd behavior can be avoided by setting the main GameObject and all children to the *Ignore Raycast* layer or modifying the `MagneticSurfaces` LayerMask array appropriately.
```
Comment (thalbern) [inline Documentation/README_Solver.md]:
```suggestion
When adding a [`SurfaceMagnetism`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SurfaceMagnetism) component to a GameObject, it is important to consider the layer of the GameObject and it's children, if any have colliders. The component works by performing various types of raycasts to determine what surface to "magnet" itself against. If the solver GameObject has a collider on one of the layers listed in the `MagneticSurfaces` property of `SurfaceMagnetism`, then the raycast will likely hit itself resulting in the GameObject attaching to it's own collider point. This odd behavior can be avoided by setting the main GameObject and all children to the *Ignore Raycast* layer or modifying the `MagneticSurfaces` LayerMask array appropriately.
```

File: Documentation/README_Solver.md [hunk 3]
```diff
@@ -156,31 +156,36 @@ To force the associated GameObject to stay vertical in any mode other than *None
...
+
+Conversely, a [`SurfaceMagnetism`](xref:Microsoft.MixedReality.Toolkit.Utilities.Solvers.SurfaceMagnetism) GameObject will not collide with surfaces on a layer not listed in the `MagneticSurfaces` property. It is generally recommended to place all desired surfaces on a dedicated layer (i.e *Surfaces*) and setting the `MagneticSurfaces` property to just this layer.  Using *default* or *everything* may result in UI components or cursors contributing to the solver.
+
+Finally, surfaces farther than the `MaxRaycastDistance` property setting will be ignored by the Surface Magnetism raycasts.
```
Comment (thalbern) [inline Documentation/README_Solver.md]:
```suggestion
Finally, surfaces farther than the `MaxRaycastDistance` property setting will be ignored by the surface magnetism raycasts.
```

Comment (Troy-Ferrell) [reply Documentation/README_Solver.md]:
```suggestion
Finally, surfaces farther than the `MaxRaycastDistance` property setting will be ignored by the `SurfaceMagnetism` raycasts.
```

**Comment by davidkline-ms** [comment]
@Troy-Ferrell, please resolve the merge conflict and re-run validation. Thanks!

**Pull request merged**